docs/upgrade-guide.md

```diff
@@ -32,6 +32,11 @@ with
 import "@luma.gl/debug";
 ```
 
+## AnimationLoop
+
+`AnimationLoop.start()` and `AnimationLoopProxy.start()` now return a `Promise` instead of a reference to itself, so you can no longer chain additional animationLoop calls after `start()`. The return `Promise` resolves when loop initialization completes.
```

Comment:
I worry that returning a promise here might be allowing users to set up complicated code flows, and we'll have to be extra careful about how we handle _and cancel_ promises returned from `start()`.

For example:

```js
const animationLoop = new AnimationLoop({
  onInitialize: () => { /* some async code that takes a few seconds */ }
})

const onLoopStart = () => { /* run some code after loop initialization */ }

animationLoop.start().then(onLoopStart)

// later on but before pageLoadPromise resolves
animation.stop() // onInitialize() does not run but onLoopStart does!

// or....

// later on but before `onInitialize()`s promise resolves
animationLoop.stop() // onLoopStart() runs even though the render loop was never started
```

The fact that calling `start()` only actually starts the render loop once two Promises resolve (each of which drop out if `stop()` has been called in the meantime with an `if (this._running)` check) just makes things extra complicated, in my view. The only reason we'd need to provide a way for a user to run code after initialization (but before the first frame renders) is not because of any work the library is doing, but because `start()` is waiting on page load and the user's own initialization logic in `onInitialize`. Something about that seems strange to me. Maybe `start()` is trying to do too much here?

If we do end up leaving these async functions in `start()`, do we need to return this promise to users? Isn't `onInitialize` enough? As far as I can tell the only things that happen between `onInitialize()` runs and `start()` resolving is: `this._initialized = true`, `this._addCallbackData()`, and the first render has been registered (though not executed) with `requestAnimationFrame`. I can't imagine that any of that would make a difference 🤔

I'm not sure I made a clear or cogent argument here, but I'm interesting in hearing if any of that made sense and what your thoughts are. :-)